src/main/java/com/iexec/sms/secret/MeasuredSecretService.java

```diff
@@ -52,27 +54,30 @@ public class MeasuredSecretService {
     private Counter initialSecretsCounter;
     private AtomicLong storedSecretsCount;
     private final Counter addedSecretsSinceStartCounter;
+    private final AtomicLong cachedSecretsCount;
+    private final LongSupplier cachedSecretsCountGetter;
 
     public MeasuredSecretService(String secretsType,
                                  String metricsPrefix,
                                  Supplier<Long> storedSecretsCountGetter,
                                  ScheduledExecutorService storageMetricsExecutorService,
-                                 int storedSecretsCountPeriod) {
+                                 int storedSecretsCountPeriod,
+                                 LongSupplier cachedSecretsCountGetter) {
```

Comment:
Should we move this new `cachedSecretsCountGetter` closer to `storedSecretsCountGetter`? As they look similar to me, they should probably be next to each other in the params list.